src/bsdfs/twosided.cpp

```diff
@@ -281,6 +281,30 @@ class TwoSidedBRDF final : public BSDF<Float, Spectrum> {
             return result;
         }
     }
+    Float eval_roughness(const SurfaceInteraction3f &si_,
+                                      Mask active) const override {
+        SurfaceInteraction3f si(si_);
+
+        if (m_brdf[0] == m_brdf[1]) {
+            si.wi.z() = dr::abs(si.wi.z());
+            return m_brdf[0]->eval_roughness(si, active);
+        } else {
+            Float result = 0.f;
+            Mask front_side = Frame3f::cos_theta(si.wi) > 0.f && active,
+                 back_side  = Frame3f::cos_theta(si.wi) < 0.f && active;
+
+            if (dr::any_or<true>(front_side))
+                result = m_brdf[0]->eval_roughness(si, front_side);
+
+            if (dr::any_or<true>(back_side)) {
+                si.wi.z() *= -1.f;
+                dr::masked(result, back_side) =
+                    m_brdf[1]->eval_roughness(si, back_side);
+            }
+
+            return result;
```

Comment:
Could we move this logic to a separate function?

Reply:
Would this be a concern of this PR? since such a refactoring is a different task than this, and should happen for other methods of this class, I would suggest doing that in another tab. Also, I am not too good at C++, and I cannot do such a refactoring without bad consequences :)

Reply:
I'm ok with this. I'll refactor it with a lambda once it's merged.